src/inputs/ObjectSelect.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import { computed } from 'mobx';
+import { inject, observer } from 'mobx-react';
+import { isBoolean, get, isObject } from 'lodash';
+import autoBindMethods from 'class-autobind-decorator';
+
+import * as Antd from 'antd';
+
+import { getNameOrDefault } from '@mighty-justice/utils';
+
+import {
+  IAntFormField,
+  IFieldConfigOptionSelect,
+  IInjected,
+  IInputProps,
+} from '../interfaces';
+
+import { getOptions } from '../utilities';
+import { IModel } from '../props';
+
+export interface IObjectSelectProps extends IInputProps {
+  keyBy: string;
+  renderOption: (option: IModel) => React.ReactNode;
+  renderSelected: (option: IModel) => string;
+}
+
+// 8 is the most number of options you can show with no scroll
+export const SHOW_OPTION_SEARCH_IF_OVER = 8;
+
+@inject('getOptions')
+@autoBindMethods
+@observer
+class ObjectSelect extends Component<IObjectSelectProps> {
+  private get injected () {
+    return this.props as IObjectSelectProps & IInjected & IInputProps & IAntFormField;
+  }
+
+  public static defaultProps: Partial<IObjectSelectProps> = {
+    keyBy: 'id',
+    renderOption: getNameOrDefault,
+    renderSelected: getNameOrDefault,
+  };
+
+  private get fieldConfig () {
+    return this.props.fieldConfig as IFieldConfigOptionSelect;
```

Comment:
This interface name looks outdated

Reply:
TBH it's complicated. I can go over it in person. Long story short it is outdated, and even the concept of an option is outdated, and I tried to fix it and this PR got 3x larger and everything broke, so I backed off.

Reply:
If you can make an issue about it, I'm happy with this for now